.travis.yml

```diff
@@ -14,13 +14,23 @@ before_script:
   - docker info
   - docker version
 script:
+  # Enable multi-architecture containers using QEMU
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   - docker buildx create --name xbuilder --use
+  # Login to Docker Hub and configure push behaviour if we are on master
   - >
     if [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
       echo "$DOCKER_PASS" | docker login -u ${DOCKER_USER} --password-stdin
       dopush="--push"
     fi
-  - docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 ${dopush} -t ${DOCKER_USER}/docker-magicmirror:latest .
+  # Build the latest stable MagicMirror release (https://stackoverflow.com/a/51761312/4934537)
+  - >
+    latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///')
+    if [ "$(docker manifest inspect bastilimbach/docker-magicmirror:"${latest_release}" > /dev/null; echo $?)" != 0 ]; then
+      docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:"${latest_release}" .
+      docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:latest .
```

Comment:
you can build the image once and then tag the image with many tags

Reply:
Oh, you are right. Good point, thanks!